src/ui/widgets/alignwidget.cpp

```diff
@@ -42,27 +42,59 @@ AlignWidget::AlignWidget(QWidget* const parent)
     mainLayout->setContentsMargins(5, 5, 5, 5);
     setLayout(mainLayout);
 
+    const auto labelsLay = new QHBoxLayout;
+    mainLayout->insertLayout(0, labelsLay);
+
+    const auto labelAlignPivot = new QLabel(tr("Align:"), this);
+    labelAlignPivot->setAlignment(Qt::AlignLeft);
+    labelsLay->addWidget(labelAlignPivot, 1);
+
+    const auto labelRelativeTo = new QLabel(tr("Relative to:"), this);
+    labelRelativeTo->setAlignment(Qt::AlignLeft);
+    labelsLay->addWidget(labelRelativeTo, 1);
+
+    labelsLay->addStretch();
+
     const auto combosLay = new QHBoxLayout;
     mainLayout->addLayout(combosLay);
 
-    //combosLay->addWidget(new QLabel(tr("Align")));
     mAlignPivot = new QComboBox(this);
     mAlignPivot->setMinimumWidth(20);
     mAlignPivot->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
     mAlignPivot->setFocusPolicy(Qt::NoFocus);
-    mAlignPivot->addItem(tr("Align Geometry"));
-    mAlignPivot->addItem(tr("Align Pivot"));
+    mAlignPivot->addItem(tr("Geometry"));
+    mAlignPivot->addItem(tr("Geometry by Pivot"));
+    mAlignPivot->addItem(tr("Pivot"));
     combosLay->addWidget(mAlignPivot);
 
-    //combosLay->addWidget(new QLabel(tr("Relative to")));
     mRelativeTo = new QComboBox(this);
     mRelativeTo->setMinimumWidth(20);
     mRelativeTo->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
     mRelativeTo->setFocusPolicy(Qt::NoFocus);
-    mRelativeTo->addItem(tr("Relative to Scene"));
-    mRelativeTo->addItem(tr("Relative to Last Selected"));
+    mRelativeTo->addItem(tr("Scene"));
+    mRelativeTo->addItem(tr("Last Selected"));
     combosLay->addWidget(mRelativeTo);
 
+    connect(mAlignPivot, QOverload<int>::of(&QComboBox::currentIndexChanged), this, [this](int index) {
+        static bool isItem2Removed = true;
+
+        if (index == 2) {
```

Comment:
hmm, "magic" numbers :)

Should probably add some defines instead of numbers, easier to figure out what's what.

I'm also not a big fan of `insertItem`, I would prefer `addItem`.

Regarding `removeItem` I think we need to get the index of what we are removing instead of using a fixed index.

Do we even need to remove anything? can't we hide/disable instead?

I can help figure out some logic to this.

Reply:
I thought you preferred indexes, no problem, I can change it.

> Do we even need to remove anything? can't we hide/disable instead?

Yes, we can hide but I'm not fan of leaving visible (even disabled) items that are never gonna be usable for an specific option. But it's just a personal UX decision, it can be done as you prefer

Reply:
Wait with this, let figure out how to manage the UI/UX first.